### PR TITLE
Add option to explicitly annotate Java frames with _[j]/_[i]

### PR DESCRIPTION
**Pull request by nitsanw**
This is not a well broken down PR, sorry.
I lumped together some code cleanup/changes here:
 * resolving many warnings pointed out by CLion
 * refactoring JVMTI deallocation
 * splitting out method name cleanup and '.' vs '/' conversion
 * annotating inlined vs real frames explicitly. This is more accurate than than the approach taken in FlameGraph/stackcollapse-perf.pl which relies on package names.
If this is too much I can split the commit into separate PRs.

With the new options I can have PMA generate this format, which let's BCC tools generate a usable collapased stack directly:
```
address size package/classname::methodname_[j];package/classname::methodname_[i]
```
One could argue that the right solution is a map file post processing script rather than further elaboration of the tool itself. Up for discussion.

### CODE REVIEW

**Comment by nitsanw** [comment]
@jrudolph any chance you can take a look?

**Comment by nitsanw** [comment]
ping?

**Comment by jrudolph** [comment]
@nitsanw sorry for the silence... busy months. I'll try to have a look soon.

**Comment by nitsanw** [comment]
@jrudolph well... at least now we know it builds!!! :-)

File: src/c/perf-map-agent.c [hunk 1]
```diff
@@ -39,10 +37,17 @@ bool unfold_simple = false;
 bool unfold_all = false;
 bool print_method_signatures = false;
 bool print_source_loc = false;
+
 bool clean_class_names = false;
+bool dotted_class_names = false;
```
Comment (jrudolph) [inline src/c/perf-map-agent.c]: Yes, makes sense to add further options. Can your document the new ones?

**Review by jrudolph — APPROVED** [review]
Sorry, it took me so long. ;)

LGTM, can you add the new options to the README?

**Comment by nitsanw** [comment]
Will update the README shortly

**Pull request merged**